main.md

```diff
@@ -193,180 +199,327 @@ Note: Applications MUST ensure that their authorization data types do not collid
+   "token_type": "example",
+   "expires_in": 3600,
+   "refresh_token": "tGzv3JOkF0XG5Qx2TlKWIA",
+   "authorization_details": [
+      {
+         "type": "https://www.someorg.com/payment_initiation",
+         "actions": [
+            "initiate",
+            "status",
+            "cancel"
+         ],
+         "locations": [
+            "https://example.com/payments"
+         ],
+         "instructedAmount": {
+            "currency": "EUR",
+            "amount": "123.50"
+         },
+         "creditorName": "Merchant123",
+         "creditorAccount": {
+            "iban": "DE02100100109307118603"
+         },
+         "remittanceInformationUnstructured": "Ref Number Merchant"
+      }
+   ]
+}
 ```
 
-## Relationship to "resource" parameter
+### Token Content {#token_content}
 
-[@I-D.ietf-oauth-resource-indicators] defines the request parameter `resource` indicating to the AS the resource(s) where the client intends to use the access tokens issued based on a certain grant.
- 
-This mechanism is a way to audience-restrict access tokens and to allow the AS to create resource specific access tokens. 
+In order to enable the RS to enforce the authorization details as approved in the authorization process, the AS MUST make this data available to the RS. 
 
-This draft can be used in conjunction with [@I-D.ietf-oauth-resource-indicators] in the same way as the `scope` parameter. The AS is supposed to narrow down the authorization details and respective permissions to the needs of the particular resource when minting an access token. 
+If the access token is a JWT [@!RFC7519], the AS is RECOMMENDED to add the `authorization_details` object, filtered to the specific audience, as top-level claim. 
 
-This depends, however, on the AS to know what authorization details are relevant for what RS. The parameter introduced in this specification can also be combined with the concept of resource indicators to make this relationship explicit. This enables the AS to narrow down the privileges of an access token to specific permissions for individual operations on specific resources (see [@I-D.ietf-oauth-security-topics], section-3.3). 
+The AS will typically also add further claims to the JWT the RS requires for request processing, e.g., user id, roles, and transaction specific data. What claims the particular RS requires is defined by the RS-specific policy with the AS.
 
-The `locations` and the `identifier` elements together allow the AS to determine the resource a client wants to access as shown in following example: 
+The following shows the contents of an example JWT payload for the payment initation example above: 
 
 ```JSON
-[  
-   {
-      "type": "https://scheme.example.org/storage":
-      "locations":["https://storage.example.com"],
-      "identifier":"/shared/group1",
-      "actions":[  
-         "read"
-      ]
-   }
+{
+   "iss": "https://as.example.com",
+   "sub": "24400320",
+   "aud": "s6BhdRkqt3",
```

Comment:
the client id used throughout the examples elsewhere in the document is the audience here?

Reply:
fixed